dbflow/src/main/java/com/raizlabs/android/dbflow/structure/ModelAdapter.java

```diff
@@ -63,12 +57,8 @@ public DatabaseStatement getInsertStatement(DatabaseWrapper databaseWrapper) {
      * @return The precompiled full statement for this table model adapter
      */
     public DatabaseStatement getCompiledStatement() {
-        if (compiledStatement == null) {
-            compiledStatement = getCompiledStatement(
-                    FlowManager.getDatabaseForTable(getModelClass()).getWritableDatabase());
-        }
-
-        return compiledStatement;
+        return getCompiledStatement(
```

Comment:
What I meant is we should delay closing it until it sits idle on the main thread. But while thats happening we should not regenerate the `SQLiteStatement` every method call here. This will improve performance such that we can rapidly save stuff without reconstructing it everytime. Itll close it after a bulk op occurs  and recreate it next time needed. I think that makes sense, right?